Sources/CryptomatorCloudAccess/GoogleDrive/GoogleDriveCloudProvider.swift

```diff
@@ -532,7 +532,7 @@ public class GoogleDriveCloudProvider: CloudProvider {
 					CloudAccessDDLogDebug("GoogleDriveCloudProvider: executeQuery(\(query.requestID)) failed with error: \(error)")
 					if error.domain == NSURLErrorDomain, error.code == NSURLErrorNotConnectedToInternet || error.code == NSURLErrorCannotConnectToHost || error.code == NSURLErrorNetworkConnectionLost || error.code == NSURLErrorDNSLookupFailed || error.code == NSURLErrorResourceUnavailable || error.code == NSURLErrorInternationalRoamingOff {
 						reject(CloudProviderError.noInternetConnection)
-					} else if error.domain == kGTLRErrorObjectDomain, error.code == 401 || error.code == 403 {
+					} else if error.domain == kGTLRErrorObjectDomain || error.domain == "org.openid.appauth.oauth_token", error.code == 401 || error.code == 403 || error.code == -10 {
```

Comment:
I would try to avoid magic strings/numbers as much as possible. Can you use `OIDOAuthTokenErrorDomain` instead of the string? And what does `-10` mean?

And I'd also like to make sure that the condition is "right". Something along the lines:

```swift
} else if ((error.domain == kGTLRErrorObjectDomain && (error.code == 401 || error.code == 403)) || (error.domain == OIDOAuthTokenErrorDomain && error.code == -10)) {
```

I just want to make sure that the domains and codes are checked together and don't get mixed up.

Reply:
You’re right! I’ve specified the error handling to better differentiate between different cases. I discovered this by analyzing the error response from AppAuth: the -10 code specifically corresponds to an “invalid_grant” error, which indicates that the token has expired or been revoked. I’ve refactored this in commit https://github.com/cryptomator/cloud-access-swift/commit/0c1a348fb7bc0d806b17528d1bf845ab85439afb.